### ISSUE TITLE
Create BOM module and extract dependencies/plugins versions into properties

### ISSUE THREAD

**Issue by hekonsek** [eclipse/hono#5]
Hi,

Good Maven practice is to keep jar dependencies in a BOM module.

Also it is also a good practice to extract dependencies and plugins version numbers into `<properties>`. This approach allows to keep versions of our libraries/plugins nicely aligned and also makes it easier to monitor updates in selected dependencies using commands like:

    mvn versions:display-property-updates

If nobody objects I will be happy to introduce BOM and extract properties.

**Comment by sophokles73** [comment]
Isn't that what you usually do in the parent POM?

**Comment by hekonsek** [comment]
BOM module allows you to separate your dependencies and public plugin configuration from internal plugins configuration. So if end user would like to use your BOM as a parent module in their project (to keep dependencies aligned with your project or use some of your pre-configured Maven plugins), they don't import the internal configuration of your project (like Maven compiler plugin settings).

So basically BOM is a way to separate public Maven configuration (used by users importing your modules) and internal project configuration (used to build your project).

See BOM section here [1] for more details on the BOM modules.

[1] https://maven.apache.org/guides/introduction/introduction-to-dependency-mechanism.html

**Comment by sophokles73** [comment]
I see, I think I have seen that approach being used by e.g. Spring Boot to provide a list of dependencies for client code to *import* by depending on it in their Maven POM.

Sounds like a good idea to me. So, by all means, feel free to add the BOM and extract properties :-)

**Comment by hekonsek** [comment]
Exactly! Spring Boot uses BOM approach :) .

I will proceed with the changes then.

**Comment by hekonsek** [comment]
Migrated project to BOM model.
